bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -81,20 +102,97 @@
 
   private final String host;
 
+  private final HeapSizeManager heapSizeManager;
   private final AsyncExecutor asyncExecutor;
+  private final ExecutorService executorService;
+  private final BigtableOptions options;
+
+  private final LinkedBlockingQueue<MutationOperation> mutationsToBeSent =
+      new LinkedBlockingQueue<>();
+
+  /**
+   * A counter for the number of {@link #mutationWorker}s are
+   * active.
+   */
+  private final AtomicInteger activeMutationWorkers = new AtomicInteger();
+
+  /**
+   * This {@link Runnable} pulls a mutation from {@link #mutationsToBeSent}, and calls {{@link
+   * #issueRequest(Mutation, long)}.
+   */
+  private final Runnable mutationWorker = new Runnable() {
+    @Override
+    public void run() {
+      activeMutationWorkers.incrementAndGet();
+      try {
+        while (!closed) {
+          try {
+            MutationOperation operation =
+                mutationsToBeSent.poll(MUTATION_TO_BE_SENT_WAIT_MS, TimeUnit.MILLISECONDS);
+            // The operation can be null if a timeout occurs.
+            if (operation != null) {
+              ListenableFuture<? extends GeneratedMessage> request =
+                  issueRequest(operation.mutation, operation.operationId);
+              ExceptionCallback callback = new ExceptionCallback(operation.mutation);
+              Futures.addCallback(request, callback);
+            }
+          } catch (InterruptedException e) {
+            Thread.interrupted();
+            LOG.info("Interrupted.  Shutting down mutationRunnable.");
+            break;
+          } catch (Exception e) {
+            LOG.error("Exception in mutation worker", e);
```

Comment:
Interrupted Exception is a signal to quit thus the break. Other exceptions shouldn't cause worker to quit.